### PR TITLE
fix(gauge):conditional gauge numberFormatter

### PR DESCRIPTION
**Pull request by RiyaJethwa**
### Updates
- Removal of default gauge numberFormatter
- Created gauge numberFormatter conditional thus when not passed, defaults to localeNumberFormatter.

### Demo screenshot or recording
When gauge numberFormatter is passed:
image

When gauge numberFormatter is not passed:
image

Default:
image

### CODE REVIEW

**Comment by RiyaJethwa** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/core/src/components/graphs/gauge.ts [hunk 1]
```diff
@@ -287,10 +294,14 @@ export class Gauge extends Component {
...
-				.text(
-					(d: any) =>
-						`${localeNumberFormatter(Number(numberFormatter(d)), localeCode)}${gaugeSymbol}`
-				)
+				.text((d: any) => {
+					const value = Number(d.toFixed(2)) % 1 !== 0 ? d.toFixed(2) : d.toFixed()
+					if (numberFormatter) {
+						return `${numberFormatter(value)}${gaugeSymbol}`
+					} else {
+						return `${localeNumberFormatter(Number(value), localeCode)}${gaugeSymbol}`
```
Comment (theiliad) [inline packages/core/src/components/graphs/gauge.ts:302]: is there ever a chance of `localeNumberFormatter` being null or undefined? we should have a fallback probably

File: packages/core/src/components/graphs/gauge.ts [hunk 2]
```diff
@@ -210,7 +210,14 @@ export class Gauge extends Component {
 			.merge(valueNumber as any)
 			.style('font-size', `${fontSize}px`)
 			.attr('text-anchor', 'middle')
-			.text((d: any) => localeNumberFormatter(Number(numberFormatter(d)), localeCode))
+			.text((d: any) => {
+				const value = Number(d.toFixed(2)) % 1 !== 0 ? d.toFixed(2) : d.toFixed()
+				if (numberFormatter) {
+					return numberFormatter(value)
+				} else {
+					return localeNumberFormatter(Number(value), localeCode)
```
Comment (theiliad) [inline packages/core/src/components/graphs/gauge.ts:218]: is there ever a chance of `localeNumberFormatter` being null or undefined? we should have a fallback probably

**Review by theiliad — CHANGES_REQUESTED** [review]

File: packages/core/src/components/graphs/gauge.ts (hunk 1 above)
Comment (RiyaJethwa) [reply packages/core/src/components/graphs/gauge.ts:302]: Do you mean the value passed to the localeNumberFormatter or the return value of localeNumberFormatter? can you please clarify.

**Review by theiliad — APPROVED** [review]

**Pull request merged**